app/src/main/java/ingsw/group1/findmyphone/activity/ModifyContactActivity.java

```diff
@@ -0,0 +1,81 @@
+package ingsw.group1.findmyphone.activity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.eis.smslibrary.SMSPeer;
+
+import ingsw.group1.findmyphone.R;
+import ingsw.group1.findmyphone.contacts.SMSContactSwipeCallback;
+import ingsw.group1.findmyphone.contacts.SMSContact;
+import ingsw.group1.findmyphone.contacts.SMSContactManager;
+
+/**
+ * Activity for adding a new contact
+ *
+ * @author Giorgia Bortoletti
+ */
+public class ModifyContactActivity extends AppCompatActivity {
+
+    private static String contactAddress;
+
+    private EditText contactNameField;
+
+    private SMSContactManager contactManager;
+    private String contactName;
+
+    /**
+     * Method invoked when activity is created
+     *
+     * @param savedInstanceStatus
+     */
+    @Override
+    public void onCreate(Bundle savedInstanceStatus) {
+        super.onCreate(savedInstanceStatus);
+        setContentView(R.layout.activity_modify_contact);
+
+        contactNameField = findViewById(R.id.modify_contact_name);
+        TextView contactPhoneField = findViewById(R.id.modify_contact_address);
+
+        contactManager = SMSContactManager.getInstance(getApplicationContext());
+
+        SMSContact contact = contactManager.getContactForPeer(new SMSPeer(contactAddress));
+        contactName = contact.getName();
+
+        contactNameField.setText(contactName);
+        contactPhoneField.setText(contactAddress);
+
+    }
+
+    /**
+     * Called when the user touches the button for modifying an existing contact
+     *
+     * @param view {@link View} where this method is invoked
+     */
+    public void modifyContact(View view) {
+        String newContactName = contactNameField.getText().toString();
+
+        if (newContactName.equals(contactName)) {
```

Comment:
Please check what happens on null text, I'm not entirely sure

Reply:
Shouldn't `contactNameField.getText()` return an empty string?

Reply:
I have tested it with empty text and the contact is updated with empty name. This is right because name is optional.